### PR TITLE
[onert/libs] Add to_underlying to misc

### PR DESCRIPTION
**Pull request by zetwhite**
This PR add to_underlying function to onert/libs/misc.
This function is to get underlying numeric value from enum types.

ONE-DCO-1.0-Signed-off-by: SeungHui Youn <sseung.youn@samsung.com> 

draft : https://github.com/Samsung/ONE/pull/12545 
in the process to resolve: https://github.com/Samsung/ONE/issues/12520

### CODE REVIEW

File: tests/tools/onert_train/src/nnfw_util.h
Comment (zetwhite) [inline tests/tools/onert_train/src/nnfw_util.h:39]: (note) to find the underlying value(usually integer) for enum
(I referred this function from modern effective c++ item 10)  
 
---------------------------------------

here are some c++ things I checked : 

* about `underlying_type_t` : 
   https://en.cppreference.com/w/cpp/types/underlying_type#:~:text=T%20%3E%0Ausing-,underlying_type_t,-%3D%20typename%20underlying_type 

* `constexpr` implies inline, that's why it defines in header file 
   https://en.cppreference.com/w/cpp/language/constexpr#:~:text=declaration%20implies%20inline

Comment (zetwhite) [reply tests/tools/onert_train/src/nnfw_util.h:39]: I have a plan to use this function here  - https://github.com/Samsung/ONE/pull/12545/files#r1470624916

File: tests/tools/onert_train/src/nnfw_util.h
Comment (Aeren1564) [inline tests/tools/onert_train/src/nnfw_util.h:45]: Is it safe to use the same name as [std::to_string](https://en.cppreference.com/w/cpp/string/basic_string/to_string)?

Comment (ragmani) [reply tests/tools/onert_train/src/nnfw_util.h:45]: It may be safe if we does not use `using namespace std`.

File: tests/tools/onert_train/src/nnfw_util.h
Comment (ragmani) [reply tests/tools/onert_train/src/nnfw_util.h:39]: How about moving this function into `runtime/libs/misc/include/misc`? Then we can also use this function in `onert`.

Comment (glistening) [reply tests/tools/onert_train/src/nnfw_util.h:39]: +1 for @ragmani.

1. I prefer to create `to_underlying.h` under `runtime/libs/misc/include/misc` just like `polymorphic_downcast.h`.
2. [SO](https://stackoverflow.com/questions/8357240/how-to-automatically-convert-strongly-typed-enum-into-int) says [`std::to_underlying`](https://en.cppreference.com/w/cpp/utility/to_underlying) is already a part of c++ since c++23.

Comment (zetwhite) [reply tests/tools/onert_train/src/nnfw_util.h]: Thanks for the detailed guide. :+1: 
Updated.

**Review by YongseopKim — APPROVED** [review]
LGTM 👍 
- about filename: I thought the file name had to be a noun such as `underlying_type` not `to_underlying` but `polymorphic_downcast`(seems verb more than noun) already is there. So I don't care the filename.

**Review by ragmani — APPROVED** [review]
LGTM

**Review by glistening — APPROVED** [review]
LGTM. I am waiting for @jyoungyun's approval as @zetwhite explicitly requested her review.

**Comment by zetwhite** [comment]
> LGTM. I am waiting for @jyoungyun's approval as @zetwhite explicitly requested her review.

Ah, At first this code was added to the `tests/onert_train`, That's why I asked her review explicitly. 
But its content updated a bit, it is okay to process without her review.

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**
